modules/creating_a_module.md

```diff
@@ -72,88 +80,51 @@ jhipsterFunc.registerModule(npmPackageName, hookFor, hookType[, callbackSubGener
-- `getAllSupportedLanguageOptions`: get all the language options supported by JHipster
-- `getAllInstalledLanguages`: get the list of languages installed by current application
-- `addChangelogToLiquibase`: add a new changelog in the Liquibase `master.xml` file for a given needle
-- `addConstraintsChangelogToLiquibase`: add a new constraints changelog to the Liquibase `master.xml` file
-- `addLiquibaseChangelogToMaster`: add a new changelog to the Liquibase `master.xml` file
-- `addColumnToLiquibaseEntityChangeset`: add new columns to the Liquibase changelog for an entity
-- `dateFormatForLiquibase`: creates a new timestamp to be used by a Liquibase changelog
-- `copyI18nFilesByName`: copy i18n files
-- `copyTemplate`: copy a template from source to a destination after stripping any translation content when translation is disabled
-- `copyHtml`: short hand method for `copyTemplate` which is defaulted to action `stripHtml`
-- `copyJs`: short hand method for `copyTemplate` which is defaulted to action `stripJs`
-- `rewriteFile`: add the given content above a specific custom needle in a file
-- `replaceContent`: replace the given content for a specific pattern/regex in a file
-- `registerModule`: register a JHipster module to act as a hook from app or entity generator
-- `getModuleHooks`: get the array of all registered module hooks for the application
-- `updateEntityConfig`: update the json configuration file for an entity with given key and value
-- `getExistingEntities`: get sorted list of entities according to changelog date in the current application
-- `isJhipsterVersionLessThan`: check if the JHipster version used to generate the project is less than a particular version
+## Available variables and functions
+
+### Variables from configuration:
 
+You have to use this function:
+
+```
+    _getConfig() {
+        const fromPath = '.yo-rc.json';
+        if (shelljs.test('-f', fromPath)) {
+            const fileData = this.fs.readJSON(fromPath);
+            if (fileData && fileData['generator-jhipster']) {
+                return fileData['generator-jhipster'];
+            }
+        }
+        return false;
+    },
+```
+
+Then, you can access to configuration in `.yo-rc.json`:
+
+```
+    this.baseName = this._getConfig().baseName;
+    this.baseName = this._getConfig().packageName;
+    this.baseName = this._getConfig().clientFramework;
+```
+
+### Global variables:
+
+You can use constants in [generator-constants](https://github.com/jhipster/generator-jhipster/blob/master/generators/generator-constants.js):
+
+```
+    const javaDir = `${constants.SERVER_MAIN_SRC_DIR + this.packageFolder}/`;
+    const resourceDir = constants.SERVER_MAIN_RES_DIR;
+    const webappDir = constants.CLIENT_MAIN_SRC_DIR;
+```
+
+### Functions:
+
+You can use all functions in [generator-base](https://github.com/jhipster/generator-jhipster/blob/master/generators/generator-base.js):
```

Comment:
We need to mention that only functions in `generato-base.js` and variables in `generator-constants.js` are part of public API and hence will follow semver. But other files like generator-base-private, utils etc will not follow semver and might break methods across minor versions